src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -201,6 +207,42 @@ public FileResource(FileService fileService, ResourceLoader resourceLoader, User
         return responseEntityForFilePath(Constants.DRAG_ITEM_FILEPATH + filename);
     }
 
+    /**
+     * GET /files/file-upload/submission/:submissionId/:filename : Get the file upload exercise submission file
+     *
+     * @param submissionId ID of the submission, the attachment belongs to
+     * @param filename  the filename of the file
+     * @return The requested file, 403 if the logged in user is not allowed to access it, or 404 if the file doesn't exist
+     */
+    @GetMapping("/files/file-upload-submission/{submissionId}/{filename:.+}")
+    @PreAuthorize("permitAll()")
+    public ResponseEntity<Resource> getFileUploadSubmission(@PathVariable Long submissionId, @PathVariable String filename) {
+        log.debug("REST request to get file : {}", filename);
+        Optional<FileUploadSubmission> optionalSubmission = fileUploadSubmissionRepository.findById(submissionId);
+        if (!optionalSubmission.isPresent()) {
+            return ResponseEntity.badRequest().build();
+        }
+        Submission submission = optionalSubmission.get();
+        try {
+            byte[] file = fileService.getFileForPath(Constants.FILE_UPLOAD_SUBMISSION_FILEPATH + submission.getId() + '/' + filename);
+            if (file == null) {
```

Comment:
This is also a duplicated fragment to lines 301-309.

Reply:
Very good point, I moved the whole `try-catch` block into a separate methods. Now, additionally it should handle properly different MIME types, not only pdf